element_calcium_imaging/scan.py

```diff
@@ -613,3 +607,125 @@ def estimate_nd2_scan_duration(nd2_scan_obj):
+
+        self.insert1(key)
+
+        for channel in range(nchannels):
+            if acq_software == "ScanImage":
+                import scanreader
+
+                movie = scanreader.read_scan(get_scan_image_files(key))[
+                    key["field_idx"], :, :, channel, :
+                ].transpose(2, 0, 1)
+            elif acq_software == "Scanbox":
+                import sbxreader
+
+                movie = sbxreader(get_scan_box_files(key))[
+                    :, key["field_idx"], channel, :, :
+                ]
+            elif acq_software == "NIS":
+                import nd2
+
+                nd2_file = nd2.ND2File(get_nd2_files(key)[0])
+
+                get_dim_pos = {k: i for i, k in enumerate(nd2_file.sizes)}
+
+                known_fields = ["T", "Z", "C", "Y", "X"]
+                fields_in_data = [x for x in known_fields if x in get_dim_pos]
+                assert len(fields_in_data) == len(
+                    get_dim_pos
+                ), f"Unknown dimension is detected in {get_dim_pos}"
+                movie = nd2_file.asarray().transpose(
+                    [get_dim_pos[x] for x in fields_in_data]
+                )
+
+                if "T" not in fields_in_data:
+                    movie = np.expand_dims(movie, 0)
+                if "Z" not in fields_in_data:
+                    movie = np.expand_dims(movie, 1)
+                if "C" not in fields_in_data:
+                    movie = np.expand_dims(movie, 2)
+
+                movie = movie[:, key["field_idx"], channel, :, :]
+
+            flat_movie = movie.reshape(movie.shape[0], -1)
+
+            self.FrameMetrics.insert1(
+                dict(
+                    key,
+                    channel=channel,
+                    min_intensity=flat_movie.min(-1),
+                    mean_intensity=flat_movie.mean(-1),
+                    max_intensity=flat_movie.max(-1),
+                    contrast=np.diff(
+                        np.percentile(flat_movie, [1, 99], axis=1), axis=0
+                    )[0],
+                )
+            )
+
+            quantalsize_results = compute_quantal_size(movie)
+            middle_frame = movie.shape[0] // 2
+            HALF_SHIFT = 250
+            HALF_SHIFT = min(HALF_SHIFT, middle_frame)
+            quantal_frame = (
+                np.mean(
+                    movie[middle_frame - HALF_SHIFT : middle_frame + HALF_SHIFT],
+                    axis=0,
+                )
+                - quantalsize_results["zero_level"]
+            ) / quantalsize_results["quantal_size"]
+
+            self.QuantalSize.insert1(
+                dict(
+                    **key,
+                    channel=channel,
+                    min_intensity=quantalsize_results["min_intensity"],
+                    max_intensity=quantalsize_results["max_intensity"],
+                    quantal_size=quantalsize_results["quantal_size"],
+                    zero_level=quantalsize_results["zero_level"],
```

Comment:
```suggestion
                    **quantalsize_results,
```

Reply:
Then we need to add `ignore_extra_fields=True` for safety.